### ISSUE TITLE
Partially consumed packet generator leaves client in a broken state

### ISSUE THREAD

**Issue by ofek** [mymarilyn/clickhouse-driver#117]
cc @xzkostyan

```
>>> from clickhouse_driver import Client
>>>
>>> client = Client(host='localhost', port=9000, user='...', password='...')
>>>
>>> client.execute("SELECT version()")
[('19.15.2.2',)]
>>>
>>> list(client.execute_iter("SELECT version()"))
[('19.15.2.2',)]
>>>
>>> list(client.execute_iter("SELECT version()"))
[('19.15.2.2',)]
>>>
>>> next(client.execute_iter("SELECT version()"))
('19.15.2.2',)
>>>
>>> next(client.execute_iter("SELECT version()"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\client.py", line 297, in execute_iter
    self.connection.force_connect()
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\connection.py", line 174, in force_connect
    elif not self.ping():
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\connection.py", line 365, in ping
    raise errors.UnexpectedPacketFromServerError(msg)
clickhouse_driver.errors.UnexpectedPacketFromServerError: Code: 102. Unexpected packet from server localhost:9000 (expected Pong, got ProfileInfo)
>>>
>>> client.execute("SELECT version()")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\client.py", line 196, in execute
    self.connection.force_connect()
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\connection.py", line 174, in force_connect
    elif not self.ping():
  File "C:\Users\ofek\AppData\Local\Programs\Python\Python37\lib\site-packages\clickhouse_driver\connection.py", line 365, in ping
    raise errors.UnexpectedPacketFromServerError(msg)
clickhouse_driver.errors.UnexpectedPacketFromServerError: Code: 102. Unexpected packet from server localhost:9000 (expected Pong, got Data)
```

**Comment by xzkostyan** [comment]
Well, yeah.

You should not emit other queries while `execute_iter` or `execute_with_progress` are not consumed all the data from wire. Same thing with [multithreading or async](https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading).

Should we raise exception here?

**Comment by ofek** [comment]
Yes please, I think so.

**Comment by ofek** [comment]
Alternatively, is there a way to force execution of a new query or clear state?

**Comment by xzkostyan** [comment]
Fixed in 0.2.3.
